uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -846,8 +869,12 @@ define([
        * Initialize user's role
        */
       function initUserRole() {
-        if (this.directiveOptions.leaveRequest &&
-          this.directiveOptions.leaveRequest.contact_id != this.directiveOptions.contactId) {
+        //first condition is if an existing request is opened by user who is not not owner of request
+        if ((this.directiveOptions.leaveRequest &&
+            this.directiveOptions.leaveRequest.contact_id != this.directiveOptions.contactId) ||
+            //request is on behalf of staff and manager has selected a staff contact
+          (this.directiveOptions.forStaff && this.contact)) {
```

Comment:
i'm not sure i understand why this whole method needs to be so complicated. We have the `roleOf()` method on `LeaveRequestInstance`, that already returns "owner", "manager", etc. Shouldn't it be something like:
```js
function initUserRole() {
  return this.request.roleOf(this.directiveOptions.contactId)
    .then(function (foo) {
      role = foo;
    });
}
```